scripts/smeshr/prepare_MODIS_HRRR_inputs.sh

```diff
@@ -0,0 +1,286 @@
+if $verbose ; then
+    echo "${tcdc_script}"
+    echo "    ${TOPO}"
+    echo "    hrrr.${YEAR}1*"
+    echo "    hrrr.t*f06.grib2"
+    echo "    ${TCDC}"
+    echo "    ${WY}"
+    echo "    0 1 2"
+    echo "    ${TCDC_out}"
+fi
+if $realrun ; then
+    ${tcdc_script} ${TOPO} "hrrr.${YEAR}1*" "hrrr.t*f06.grib2" ${TCDC} ${WY} "0 1 2" ${TCDC_out}
+fi
+# Run January through September for your WY
+if $verbose ; then
+    echo "${tcdc_script}"
+    echo "    ${TOPO}"
+    echo "    hrrr.${WY}*"
+    echo "    hrrr.t*f06.grib2"
+    echo "    ${TCDC}"
+    echo "    ${WY}"
+    echo "    3 4 5 6 7 8 9 10 11"
+    echo "    ${TCDC_out}"
+fi
+if $realrun ; then
+    ${tcdc_script} ${TOPO} "hrrr.${WY}0*" "hrrr.t*f06.grib2" ${TCDC} ${WY} "3 4 5 6 7 8 9 10 11" ${TCDC_out}
+fi
+
+echo ; echo " ============== Finished processing HRRR cloud cover ============== " ; echo 
+
+#==========================================
+#========== Process SW radiation ==========
+#==========================================
+# Create directory for SW outputs
+if $verbose ; then
+    echo "mkdir -pv ${SMESHR_DIR}${BASIN}/DSWRF/" ; echo
+fi
+if $realrun ; then
+    if [ ! -d ${SMESHR_DIR}${BASIN}/DSWRF/ ] ; then
+        mkdir -pv ${SMESHR_DIR}${BASIN}/DSWRF/
+    fi
+fi
+
+# Extract SW from HRRR outputs 
+# Run for the WY, looping month by month beginning with the last day of the preceding September
+dt=${YEAR}09
+end_dt=${WY}09
+
+while [ "$dt" -le "$end_dt" ]
+do
+    if $verbose ; then
+        echo -ne "Processing $dt\r"
+        sleep 0.01
+    fi
+    echo ${dt}
+    if $realrun ; then
+        # check for preceding september and process last day only
+        if [ ${dt} == ${YEAR}09 ] ; then
+            ${dswrf_script} ${TOPO} "hrrr.${dt}30" "hrrr.t*f06.grib2" ${DSWRF_DIR}
+        else
+            ${dswrf_script} ${TOPO} "hrrr.${dt}*" "hrrr.t*f06.grib2" ${DSWRF_DIR}
+        fi
+    fi
+    dt=$(date -d "${dt}01 +1 month" +%Y%m)
+done
+
+echo ; echo ; echo " ============== Finished processing HRRR SW radiation ============== " ; echo
+
+#==========================================
+#========== Process MODIS albedo ==========
+#==========================================
+# Edit MODIS_albedo_basin.sh
+function update_verbose()
+{
+    echo "          Formerly: $1"
+    echo "          Now: $2"
+    sed -i "s|${1}|${2}|g" $3 
+    echo
+}
+
+# Extract extents of basin from topo.nc
+# Get x bounds
+minx=$(cdo sinfon ${TOPO} | grep x | tail -n 1 | cut -d ':' -f2 | cut -d ' ' -f2)
+maxx=$(cdo sinfon ${TOPO} | grep x | tail -n 1 | cut -d ':' -f2 | cut -d ' ' -f4)
+
+# Get y bounds. Note that this order is reversed in cdo sinfon output
+miny=$(cdo sinfon ${TOPO} | grep y | tail -n 1 | cut -d ':' -f2 | cut -d ' ' -f4)
+maxy=$(cdo sinfon ${TOPO} | grep y | tail -n 1 | cut -d ':' -f2 | cut -d ' ' -f2)
+
+# Adjust extents based on half the resolution size (RES) and basic calculator (bc)
+# Decrease minx and miny
+minx=$(echo "${minx} - ${RES}/2" | bc)
```

Comment:
For future bash math - one can do that without the use of a calculator. Demo:
```bash
$ export a="1"
$ export b="2"
$ echo "$(($a + $b))"
3
```

This script is coming together nicely! Are you a bash shell scripting fan yet? :laughing:

Reply:
Thank you! I think I can only say that I am a true fan of shared meals and communal cooking. But shell scripting is certainly powerful and for that, I very much appreciate it 😄